tracer/src/Datadog.Trace/Ci/Coverage/Metadata/ModuleCoverageMetadata.cs

```diff
@@ -18,15 +19,45 @@ public abstract class ModuleCoverageMetadata
     /// Gets or sets the metadata array
     /// </summary>
 #pragma warning disable SA1401
-    protected readonly int[][] Metadata = Array.Empty<int[]>();
+    protected readonly long[] Metadata = Array.Empty<long>();
+#pragma warning restore SA1401
+
+    /// <summary>
+    /// Gets or sets the metadata array
+    /// </summary>
+#pragma warning disable SA1401
+    protected readonly int[] SequencePoints = Array.Empty<int>();
+#pragma warning restore SA1401
+
+    /// <summary>
+    /// Gets the total instructions number
+    /// </summary>
+#pragma warning disable SA1401
+    public readonly long TotalInstructions;
 #pragma warning restore SA1401
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal int GetTotalTypes() => Metadata.Length;
+    internal int GetMethodsCount() => Metadata.Length;
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal int GetTotalMethodsOfType(int type) => Metadata[type].Length;
+    internal int GetTotalSequencePointsOfMethod(int methodIndex)
+    {
+#if NETCOREAPP3_0_OR_GREATER
+        return SequencePoints.FastGetReference(methodIndex);
+#else
+        return SequencePoints[methodIndex];
+#endif
```

Comment:
Given `FastGetReference` is aggressively inlined + has an `#else` clause that just uses the indexer, can prob remove the `#if`s here?

```suggestion
        return SequencePoints.FastGetReference(methodIndex);
```

Reply:
I tried but the `ref` keyword generates more assembly code than a normal indexer (and we don't need the ref in this case).